scipy/special/_support_alternative_backends.py

```diff
@@ -133,16 +136,18 @@ def __betaincc(a, b, x):
 
 
 def _stdtr(xp, spx):
-    betainc = getattr(spx, 'betainc', None)  # noqa: F811
+    if spx is None:
+        return None
+    betainc = getattr(spx.special, 'betainc', None)  # noqa: F811
     if betainc is None and hasattr(xp, 'special'):
         betainc = getattr(xp.special, 'betainc', None)
     if betainc is None:
         return None
 
     def __stdtr(df, t):
         x = df / (t ** 2 + df)
-        tail = betainc(df / 2, xp.asarray(0.5), x) / 2
-        return xp.where(x < 0, tail, 1 - tail)
+        tail = betainc(df / 2, 0.5, x) / 2
```

Comment:
As is, this would upcast results to the default floating point type, which would not respect type.

Technically, we should do something like:
```python3
dtype = xp.result_type(df, t)
xp.asarray(0.5, dtype=dtype)
```
because the standard does not allow Python floats as inputs to `xp.where`. But `array_api_strict` doesn't get here, and every other backend tries to be useful, so let's not burden ourselves at this time.

Reply:
I would be in favour of making that change anyway - even if not necessary now, I think it's better to use the patterns which we'd like to see elsewhere. Not blocking, but if you're pushing other changes...

Reply:
> I think it's better to use the patterns which we'd like to see elsewhere.

Great, in that case I'd like to leave it - I would not like to see the technically correct (ATM) pattern everwhere because I think there are strong winds to make the standard more reasonable (data-apis/array-api#807).

Reply:
Okay